Tavisca.Bootcamp.LanguageBasics.Exercise1/Program.cs

```diff
@@ -14,16 +14,64 @@ static void Main(string[] args)
             Console.ReadKey(true);
         }
 
-        private static void Test(string args, int expected)
-        {
+        private static void Test(string args, int expected)
+        {
             var result = FindDigit(args).Equals(expected) ? "PASS" : "FAIL";
-            Console.WriteLine($"{args} : {result}");
+            Console.WriteLine($"{args} : {result}");
+        }
+
+        public static int FindDigit(string equation)
+        {
+            /* Splitting the equation into three parts A,B,C */
+
+            string[] token = equation.Split("=");
+            string[] equ = token[0].Split("*");
+            string a = equ[0],b=equ[1],c=token[1];
+            int an,bn,cn;
+
+            /*First checking if C contains ? character */
+
+            if(c.Contains("?")){
+                an = Convert.ToInt32(a);
+                bn = Convert.ToInt32(b);
+                string cnew= Convert.ToString(an*bn);
+
+                /* findingDigit method will compare two strings  for different characters or length */
+                return findingDigit(cnew,c);
+            }
+
+            /*Checking for A only if it contains '?' if B contains then swap B with A */
+
+            else{
+                if(b.Contains("?")){
+                    string t= a;
+                    a=b;
+                    b=t;
+                }
+                cn = Convert.ToInt32(c);
+                bn = Convert.ToInt32(b);
+                if(cn%bn!=0 || bn==0)
+                    return -1;
+                string anew= Convert.ToString(cn/bn);
+                
+                 /* findingDigit method will compare two strings  for different characters or length */
+                return findingDigit(anew,a);
+            }
         }
 
-        public static int FindDigit(string equation)
-        {
-            // Add your code here.
-            throw new NotImplementedException();
+        public static int findingDigit(string st,string c){

```

Comment:
Method name should always start with upper case letter.
Also you should use Verb in it's first form in method name, so it should be "FindDigit" and not "FindingDigit".

Also try to keep given method signature same. Try using FindDigit(string equation) and do your code in that method.

Also 'st' and 'c'...??? Please have some meaningful names